### PR TITLE
FAI-14158: Azure WorkItems Hardening

### PR DESCRIPTION
**Pull request by chalenge**
## Description

- Add changelogs for assignees and states
- Add state mapping
- Add type mapping
- More fixes for tms_Task mappings

## Type of change
- [X] Bug fix
- [X] New feature
- [ ] Breaking change

### CODE REVIEW

File: destinations/airbyte-faros-destination/src/converters/azure-workitems/workitems.ts [hunk 1]
```diff
@@ -15,45 +21,108 @@ export class Workitems extends AzureWorkitemsConverter {
...
     return [
       {
         model: 'tms_Task',
         record: {
-          uid: String(WorkItem.id),
-          id: String(WorkItem.id),
-          url: WorkItem.url,
-          type: {category: String(WorkItem.fields['System.WorkItemType'])},
+          ...taskKey,
+          id: taskKey.uid,
```
Comment (ypc-faros) [inline destinations/airbyte-faros-destination/src/converters/azure-workitems/workitems.ts:45]: Remove. `id` is autogen

File: destinations/airbyte-faros-destination/src/converters/azure-workitems/workitems.ts [hunk 2]
```diff
@@ -15,45 +21,108 @@ export class Workitems extends AzureWorkitemsConverter {
...
+          url: WorkItem._links?.html?.href || WorkItem.url,
+          type: this.getTaskType(WorkItem.fields['System.WorkItemType']),
           name: WorkItem.fields['System.Title'],
-          createdAt: new Date(WorkItem.fields['System.CreatedDate']),
+          createdAt: Utils.toDate(WorkItem.fields['System.CreatedDate']),
           parent: {uid: String(WorkItem.fields['System.Parent']), source},
```
Comment (ypc-faros) [inline destinations/airbyte-faros-destination/src/converters/azure-workitems/workitems.ts:50]: Is it guaranteed to have a parent?

File: sources/azure-workitems-source/resources/spec.json [hunk 3]
```diff
@@ -14,12 +14,25 @@
...
+          "type": "string"
+        },
+        "title": "Project IDs or project names",
+        "description": "Azure Projects. If empty or '*' all available projects will be processed."
       },
       "project": {
         "type": "string",
-        "title": "Azure Project"
+        "title": "Azure Project",
+        "deprecated": true,
```
Comment (ypc-faros) [inline sources/azure-workitems-source/resources/spec.json:34]: Let's just remove?

**Review by ypc-faros — APPROVED** [review]

**Comment by chalenge** [comment]
@Christopher-Haumann FYI, making significant changes that will impact how you're using it. Not sure if you using a specific version or always latest.

File: sources/azure-workitems-source/resources/spec.json (hunk 3 above)
Comment (chalenge) [reply sources/azure-workitems-source/resources/spec.json:34]: Didn't wanna making it a breaking change

**Pull request merged**